app/models/topic_tracking_state.rb

```diff
@@ -217,14 +245,27 @@ def self.publish_read(topic_id, last_read_post_number, user_id, notification_lev
 
   def self.publish_dismiss_new(user_id, topic_ids: [])
     message = {
-      message_type: "dismiss_new",
+      message_type: DISMISS_NEW_MESSAGE_TYPE,
       payload: {
         topic_ids: topic_ids
       }
     }
     MessageBus.publish(self.unread_channel_key(user_id), message.as_json, user_ids: [user_id])
   end
 
+  def self.new_filter_sql
+    TopicQuery.new_filter(Topic, "xxx").where_clause.ast.to_sql.gsub!("'xxx'", treat_as_new_topic_clause) +
+      " AND topics.created_at > :min_new_topic_date" +
+      " AND dismissed_topic_users.id IS NULL"
+  end
+
+  def self.unread_filter_sql(staff: false)
+    TopicQuery
+      .unread_filter(Topic, -999, staff: staff)
+      .where_clause.ast.to_sql
+      .gsub("-999", ":user_id")
```

Comment:
Same here - I don't love gsub on SQL with placeholder values.

Reply:
Bizarrely this was not even used -- the `unread_filter` didn't even have a `user_id` as part of its parameters. Fixed now.